switchapi/es2k/switch_lag.c

```diff
@@ -296,36 +362,163 @@ switch_status_t switch_api_lag_update(const switch_device_t device,
     SWITCH_ASSERT(status == SWITCH_STATUS_SUCCESS);
 
-    // create rx path
-    status = switch_pd_rx_lag_table_entry(device, lag_info, true);
+  } else if (lag_info->num_active_ports == 1) {
+    /* oper_state false means num_active_ports is decremented from 2 to 1
+     and oper_state true means it is incremented from 0 to 1 */
+    if (oper_state) {
+      // populate the tx and rx entries with single active member
+      status = SWITCH_LIST_INSERT(&lag_info->lag_members,
+                                  &lag_member_info->node, lag_member_info);
+      SWITCH_ASSERT(status == SWITCH_STATUS_SUCCESS);
+      status = switch_pd_tx_lacp_lag_table_entry(device, lag_info, true);
+      if (status != SWITCH_STATUS_SUCCESS) {
+        krnlmon_log_error(
+            "Failed to create tx_lacp_lag_table entry on device %d: "
+            "error: %s\n",
+            device, switch_error_to_string(status));
+        return status;
+      }
+
+      status = switch_pd_rx_lacp_lag_table_entry(device, lag_info, true);
+      if (status != SWITCH_STATUS_SUCCESS) {
+        krnlmon_log_error(
+            "Failed to create rx_lacp_lag_table entry on device %d: "
+            "error: %s\n",
+            device, switch_error_to_string(status));
+        return status;
+      }
+    } else {
+      // delete the existing entries and populate the tx and rx entries with
+      // other active member
+      status = switch_pd_tx_lacp_lag_table_entry(device, lag_info, false);
+      if (status != SWITCH_STATUS_SUCCESS) {
+        krnlmon_log_error(
+            "Failed to delete tx_lacp_lag_table entry on device %d: "
+            "error: %s\n",
+            device, switch_error_to_string(status));
+        return status;
+      }
+
+      status = switch_pd_rx_lacp_lag_table_entry(device, lag_info, false);
+      if (status != SWITCH_STATUS_SUCCESS) {
+        krnlmon_log_error(
+            "Failed to delete rx_lacp_lag_table entry on device %d: "
+            "error: %s\n",
+            device, switch_error_to_string(status));
+        return status;
+      }
+
+      status =
+          SWITCH_LIST_DELETE(&lag_info->lag_members, &lag_member_info->node);
+      SWITCH_ASSERT(status == SWITCH_STATUS_SUCCESS);
+
+      status = switch_pd_tx_lacp_lag_table_entry(device, lag_info, true);
+      if (status != SWITCH_STATUS_SUCCESS) {
+        krnlmon_log_error(
+            "Failed to create tx_lacp_lag_table entry on device %d: "
+            "error: %s\n",
+            device, switch_error_to_string(status));
+        return status;
+      }
+
+      status = switch_pd_rx_lacp_lag_table_entry(device, lag_info, true);
+      if (status != SWITCH_STATUS_SUCCESS) {
+        krnlmon_log_error(
+            "Failed to create rx_lacp_lag_table entry on device %d: "
+            "error: %s\n",
+            device, switch_error_to_string(status));
+        return status;
+      }
+    }
+  } else if (lag_info->num_active_ports == 2) {
```

Comment:
maybe we should capture this limiataion, we are supporting only 2 LAG members as of now.

Reply:
Captured in the FSD.